samples/drivers/watchdog/src/main.c

```diff
@@ -83,6 +83,23 @@ static void wdt_callback(const struct device *wdt_dev, int channel_id)
 }
 #endif /* WDT_ALLOW_CALLBACK */
 
+#if WDT_EARLY_WARNING_INTERRUPT_SUPPORT
+
+static void wdt_ew_callback(const struct device *wdt_dev, int channel_id)
```

Comment:
if the devictree bool option is valid, we can select the correct callback inspecting that devicetree option.